pkg/storage/bucket/s3/config_test.go

```diff
@@ -62,6 +62,43 @@ func TestSSEConfig_Validate(t *testing.T) {
 	}
 }
 
+func TestConfig_Validate(t *testing.T) {
+	tests := map[string]struct {
+		setup    func() *Config
+		expected error
+	}{
+		"should pass with default config": {
+			setup: func() *Config {
+				sseCfg := &SSEConfig{}
+				flagext.DefaultValues(sseCfg)
+				cfg := &Config{
+					Endpoint:         "s3.eu-central-1.amazonaws.com",
+					BucketName:       "mimir-block",
+					SSE:              *sseCfg,
+					SignatureVersion: SignatureVersionV4,
+				}
+				return cfg
+			},
+		},
+		"should fail on invalid endpoint prefix": {
+			setup: func() *Config {
+				return &Config{
+					Endpoint:         "mimir-blocks.s3.eu-central-1.amazonaws.com",
+					BucketName:       "mimir-block",
```

Comment:
This is interesting because shows a sort of flaw in the check. The bucket name is NOT the exact endpoint prefix. I think you should improve the check (`strings.HasPrefix()`) to also look for the dot (`.`) at the end of the bucket name when checking it.